### ISSUE TITLE
Odd warning messages displayed in 2.3 when supplying a provider dictionary for any eos_* module

### ISSUE THREAD

**Issue by dathelen** [ansible/ansible#24978]
##### ISSUE TYPE
 - Bug Report
 

##### COMPONENT NAME
Any eos_* module when you supply a provider var to the playbok

##### ANSIBLE VERSION

```
ansible --version
ansible 2.3.0.0
  config file = /Users/ad/Documents/src/ansible-test/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Apr  4 2017, 08:46:44) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### OS / ENVIRONMENT
Any

##### SUMMARY
When running a playbook in Ansible 2.3 the following warnings are displayed everytime a provider var is supplied:

```
[WARNING]: argument use_ssl has been deprecated and will be removed in a future version

 [WARNING]: argument port has been deprecated and will be removed in a future version

 [WARNING]: argument validate_certs has been deprecated and will be removed in a future version
```

Here is the provider var example to see the issue:
```
eapi_conn:
      host: '{{ inventory_hostname }}'
      username: admin
      password: admin
      transport: eapi
      use_ssl: true
      validate_certs: false
````

>

##### EXPECTED RESULTS
A clean run with the warnings.

##### ACTUAL RESULTS
Are these warnings accurate?  Are they being depreciated? The documentation doesn't seem to indicate that these are going away and if they are what is the correct to set the validate_certs and use_ssl attributes? 

```

```

**Comment by jctanner** [comment]
cc @privateip @trishnaguha

**Comment by trishnaguha** [comment]
@dathelen These arguments when used in `provider` should not throw deprecation warning.
 I am unable to reproduce this issue fully. This issue should not exist as per the logic in here https://github.com/ansible/ansible/blob/stable-2.3/lib/ansible/module_utils/eos.py#L76.

I was able to reproduce deprecation warning only for `transport` using in provider which should be fixed by #25782 .
With this fix enabled I do not get any deprecation warning while using the params in provider.

**Comment by trishnaguha** [comment]
@dathelen please upgrade your ansible version to `2.3.1` version as it was added in 2.3.1.
